Cue/app/tabs/library/DeckSharingOptions.js

```diff
@@ -0,0 +1,391 @@
+import CueHeader from '../../common/CueHeader'
+import CueIcons from '../../common/CueIcons'
+
+import SelectableTextTableRow from '../../common/SelectableTextTableRow'
+import TableHeader from '../../common/TableHeader'
+import ShareCodeTableRow from './ShareCodeTableRow'
+import TagsTableRow from './TagsTableRow'
+import TableFooter from '../../common/TableFooter'
+
+const styles = {
+  container: {
+    flex: 1,
+    backgroundColor: Platform.OS === 'android' ? 'white' : CueColors.coolLightGrey,
+  },
+}
+
+type Props = {
+  navigator: Navigator,
+  deck: Deck,
+
+  // From Redux:
+  editTags: (deckUuid: string, newTags: Array<string>) => any,
+  recordShareCode: (deckUuid: string, shareCode: string) => any,
+  setPublic: (deckUuid: string, isPublic: boolean) => any,
+}
+
+type SharingStatus = 'private' | 'shared' | 'public'
+
+class DeckSharingOptions extends React.Component {
+  props: Props
+  state: {
+    dirty: boolean,
+    editable: boolean,
+    selectedOption: SharingStatus,
+    shareCode: string,
+    tags: Array<string>,
+  }
+
+  // A reference to the text input inside TagsTableRow, used so
+  // KeyboardAwareScrollView knows how far to scroll to keep it visible
+  newTagTextInputRef: any
+
+  constructor(props: Props) {
+    super(props)
+
+    this.state = {
+      dirty: false,
+      editable: props.deck.accession ? props.deck.accession === 'private' : false,
+      selectedOption: this._sharingStatusForDeck(props.deck),
+      shareCode: props.deck.share_code,
+      tags: props.deck.tags || [],
+    }
+
+    this.newTagTextInputRef = null
+  }
+
+  _onTagsTableRowRef = (ref: TagsTableRow) => {
+    this.newTagTextInputRef = ref ? ref.textInputRef : null
+  }
+
+  _sharingStatusForDeck = (deck: Deck): SharingStatus => {
+    return deck.public
+      ? 'public'
+      : deck.share_code
+        ? 'shared'
+        : 'private'
+  }
+
+  _arraysEqual = (a: ?Array<*>, b: ?Array<*>): boolean => {
+    if (!a || !b) {
+      return false
+    }
+    
+    if (a.length !== b.length) {
+      return false
+    }
+
+    for (let i = 0; i < a.length; i++) {
+      if (a[i] !== b[i]) {
+        return false
+      }
+    }
+
+    return true
+  }
+
+  /* ==================== Edit Flow ==================== */
+  _onSubmit = () => {
+    if (this.state.selectedOption === 'public') {
+      this._handlePublicSubmit()
+    } else if (this.state.selectedOption === 'shared') {
+      this._handleSharedSubmit()
+    } else if (this.state.selectedOption === 'private') {
+      this._handlePrivateSubmit()
+    }
+  }
+
+  _handlePublicSubmit = () => {
+    let oldTags = this.props.deck.tags
+    let newTags = this.state.tags
+
+    if (!this._arraysEqual(oldTags, newTags)) {
+      this.props.editTags(this.props.deck.uuid, newTags)
+    }
+
+    if (!this.props.deck.public) {
+      this.props.setPublic(this.props.deck.uuid, true)
+    }
+
+    this.props.navigator.pop()
+  }
+
+  _handleSharedSubmit = () => {
```

Comment:
I'm not sure, it depends on what the actual behaviour would be... Do the decks disappear from Libraries? Are they automatically promoted to Shared? #117?